Added --exclude-networks switch.

xliiv opened the pull request:
This switch allows to exclude network from exported networks.
In addition networking resources was updated.

mkurek commented on src/ralph/export_to_ng/management/commands/export_model.py:
```diff
@@ -125,6 +141,11 @@ class Command(BaseCommand):
         ),
+        make_option(
+            '--exclude-networks',
+            help='Excludes networks from passed range, eg.: --exclude-networks=<min-ip>,<max-ip>', # noqa
```
min and max ips are "ok", but IMO it'd be better to pass network with mask here (ex. 10.20.30.00/24).
Btw what if I want to exclude many networks? Consider `action="append"` here

mkurek commented on src/ralph/export_to_ng/resources.py:
```diff
@@ -430,6 +430,12 @@ def get_queryset(self):
 
+    def get_queryset(self):
+        queryset = models_network.Network.objects.exclude(
+            min_ip__gte=self.min_ip, max_ip__lte=self.max_ip,
```
what if `min_ip` and `max_ip` are not defined?

vi4m commented on src/ralph/export_to_ng/management/commands/export_model.py:
```diff
@@ -109,6 +110,21 @@ def get_resource(model_name):
+        try:
+            min_ip, max_ip = options['exclude_networks'].split(',')
+        except ValueError:
+            print(
```
Print errors not on stdout, but stderr.

ghost commented:
TODO: test if data are importable in NG

szok commented:
:ok:

mkurek commented on src/ralph/export_to_ng/management/commands/export_model.py:
```diff
@@ -125,8 +131,33 @@ class Command(BaseCommand):
+                    )
+                    sys.exit()
+                excluded_networks.append(network)
+        model_resource.excluded_networks = excluded_networks
```
shouldn't it be set only on Network model?

xliiv replied:
of course :) thanks

mkurek commented on src/ralph/export_to_ng/resources.py:
```diff
@@ -429,6 +430,16 @@ def get_queryset(self):
 
 class NetworkResource(resources.ModelResource):
```
I'd add here `excluded_networks = []` property if somehow this resource is called outside of mgmt command

Commit:
Added --exclude-networks switch.

This switch allows to exclude network from exported networks.
In addition networking resources was updated.

mkurek commented on src/ralph/export_to_ng/resources.py:
```diff
@@ -30,11 +30,29 @@
+    """
+    try:
+        data_center_id = DataCenter.objects.get(
+            name__iexact=network_datacenter.name
```
couldn't you use `deprecated_ralph_dc` field (https://github.com/allegro/ralph_assets/blob/develop/src/ralph_assets/models_dc_assets.py#L128)?

xliiv replied:
Could you be more descriptive? :)

mkurek commented on src/ralph/export_to_ng/resources.py:
```diff
@@ -427,8 +445,61 @@ def get_queryset(self):
+        model = models_network.NetworkKind
+
+
+class DiscoveryDataCenterResource(resources.ModelResource):
```
shouldn't we skip here DCs which have equivalent in assets? comment is suggesting that, but i don't see any code that is doing that

xliiv replied:
I think this docsting is obsolete.
Something more like this
```
This exports only data centers from:
models_network.DataCenter
```
is correct but in such case there is no point to put it at all, imho.

xliiv replied:
Done

szok commented:
:ok:

mkurek commented:
:ok:
